test/app.test.ts

```diff
@@ -12,22 +12,27 @@ DataLakeService.prototype.listFiles = jest
 
 DataLakeService.prototype.uploadFile = jest.fn();
 
-beforeAll(async () => {
-    await connectToDb(datasourceOptions);
-    const datafile1 = new Datafile();
-    datafile1.name = 'test-data-1.csv';
-    datafile1.description = 'Test Data File 1';
-    datafile1.id = 'bdc40218-af89-424b-b86e-d21710bc92f1';
-    await datafile1.save();
-    const datafile2 = new Datafile();
-    datafile2.name = 'test-data-2.csv';
-    datafile2.description = 'Test Data File 2';
-    datafile2.id = 'fa07be9d-3495-432d-8c1f-d0fc6daae359';
-    await datafile2.save();
-    console.log(`Datafile created: ${Datafile.find()}`);
-});
-
 describe('Test app.ts', () => {
+    beforeAll(async () => {
+        await connectToDb(datasourceOptions);
+        await dbManager.initializeDataSource();
+
+        const dataset1 = Dataset.createDataset('Test Data 1', 'test', 'bdc40218-af89-424b-b86e-d21710bc92f1');
+        await dataset1.save();
+        dataset1.live = true;
+        dataset1.code = 'tst0001';
+        dataset1.addTitleByString('Test Dataset 1', 'EN');
+        dataset1.addDescriptionByString('I am the first test dataset', 'EN');
```

Comment:
Missing call to write back the changes?

Reply:
The functions have calls to persist the data.  This screenshot shows a Dataset that's been changed into a DTO after it has first been retrieved back from the database.  As you can see buried in the JSON are the title and description entires.

<img width="1530" alt="image" src="https://github.com/Marvell-Consulting/statswales-backend/assets/29251905/a6015458-12b8-4e5f-8892-a4353e7af84b">

I can add some tests around this if needed.... I need to expand the test base in a later PR anyway.